# Alan Almaraz 03/09/2024

## EJERCICIO 1
El perímetro de un triángulo isósceles es de 20 cm. Imagina que este triángulo rota centrado en su eje de simetría para formar la ilusión de un cono.

¿Cuánto deben medir sus lados para que el volumen generado por la rotación sea el mayor posible?

In [75]:
import sympy as sp
import math as m

def perimetro(a, b):
    return 2*a + b - 20

def pitagoras(a, b, h):
    return (b/2)**2 + h**2 - a**2

def volumen(h):
    return (1/3)*m.pi*h**2
    
p_ = sp.solve(perimetro(a, b), b)
pt_ = pitagoras(a, b, h).subs(b, p_[0])
h_ = sp.solve(pt_, h)[0]
v = volumen(h).subs(h, h_)
dv = sp.diff(v, a)
dv

20.9439510239320

Los lados debe de medir aproximadamente 20.94

## EJERCICIO 2
Disponemos de una barra de aluminio de 6 metros para construir una portería de fútbol en un parque. Es una portería simple que tendrá sólo dos postes y el travesaño:

¿Si queremos que el área de la portería sea máxima, ¿cuánto deben medir los postes y el travesaño?

In [97]:
def restricción(x, y):
    return 2*x + y - 6

def área(x, y):
    return x*y
sp.var('y')
y_ = sp.solve(restricción(x, y), y)
a_ = área(x, y).subs(y, y_[0])
da = sp.diff(a_, x)
puntos_criticos = sp.solve(da, x)
puntos_criticos

[3/2]

El punto crítico está en 1.5 metros lo que significa que eso es lo que debe medir cada poste y el atravesaño por lo tanto debe medir 3 metros

## EJERCICIO  3

In [114]:
def restricción(w, h):
    return w**2 + h**2 - 24**2

def resistencia(w, h, k):
    return k*w*(h**2)

sp.var('w')
sp.var('k')
h2 = sp.solve(restricción(w, h), h**2)[0]
r = resistencia(w, h, k).subs(h**2, h2)
dr = sp.diff(r, w)
pc = sp.solve(dr, w)[1]
pc

8*sqrt(3)

Al momento de sacar punto crítico nos dio $$ 8\sqrt3 $$ por lo que en este se obtiene la máxima resistencia posible

## Ejercicio 4

In [134]:
import sympy as sp
def I(a, s):
    return (k*sp.sin(a))/(s**2)
def pitagoras(s, h):
    return h**2 + 4 - s**2

sp.var('s')
s2 = sp.solve(pitagoras(s, h), s**2)[0]
I_ = I(a,s).subs(s**2, s2)
sp.solve(pitagoras(s, h), h)[0]
I_

k*sin(a)/(h**2 + 4)

In [180]:
import sympy as sp
def I(a, s):
    return (k*sp.sin(a))/(s**2)
def pitagoras(s, h):
    return h**2 + 4 - s**2

sp.var('s')
s_a = h / s
I_ = I(a,s).subs(sp.sin(a), s_a)
s2 = sp.solve(pitagoras(s, h), s**2)[0]
I_2 = I_.subs(s, s2**0.5)
di = sp.diff(I_2, h)
pc = sp.solve(di, h)[1]
pc



1.41421356237310

La altura máxima para que la iluminación sea la máxima sería 1.41

## EJERCICIO  5

In [ ]:
def dirección_v(n, w, Fy):
    return n - w - Fy

def dirección_h(Fx, f):
    return f - Fx


